components/map/map.jsx

```diff
@@ -10,61 +16,146 @@ const POKE_MARKERS = [
-      scrollwheel: false,
+      panControl: true,
+      draggable: true,
+      scrollwheel: true,
+    });
+    const infoMarker = new google.maps.Marker({
+      position: { lat: 35, lng: -95 },
+      icon: POKEBALL_MARKERS[4],
+      map: map
+    });
+    const infoWindow = new google.maps.InfoWindow({
+      content: "<div style='width: 212px; font-size: 10px; margin-top: 5px;'>" +
+                 "Click on pokeballs to search for pokemon!" +
+               "</div>"
+    });
+
+    infoWindow.open(map, infoMarker);
+
+    infoMarker.addListener('click', () => {
+      infoMarker.setMap(null);
+      this.addPokeballMarker(map);
+
+      const markerLat = infoMarker.getPosition().lat();
+      const markerLng = infoMarker.getPosition().lng();
+
+      map.panTo({ lat: markerLat, lng: markerLng });
+      map.setZoom(10);
+
+      this.addPokemonMarker(map, markerLat, markerLng);
     });
 
-    for (var i = 0; i < 25; i++) {
-      this.addMarker(map);
+    for (let i = 0; i < 750; i++) {
+      this.addPokeballMarker(map);
     }
+
+    this.setState({
+      map: map,
+      pokemon: null,
+      modalOpen: false
+    });
   }
 
-  addMarker(map) {
+  addPokeballMarker(map) {
+    const randomPokeball = POKEBALL_MARKERS[Math.floor(Math.random() * POKEBALL_MARKERS.length)];
     const lat = Math.random() * (MAX_LAT - MIN_LAT) + MIN_LAT;
     const lng = Math.random() * (MAX_LNG - MIN_LNG) + MIN_LNG;
 
-    const marker = new google.maps.Marker({
-      position: {lat, lng},
-      icon: POKE_MARKERS[Math.floor(Math.random() * POKE_MARKERS.length)],
+    const pokeballMarker = new google.maps.Marker({
+      position: { lat, lng },
+      icon: randomPokeball,
+      map: map
+    });
+
+    pokeballMarker.addListener('click', () => {
+      pokeballMarker.setMap(null);
+      this.addPokeballMarker(map);
+
+      const markerLat = pokeballMarker.getPosition().lat();
+      const markerLng = pokeballMarker.getPosition().lng();
+
+      map.panTo({ lat: markerLat, lng: markerLng });
+      map.setZoom(10);
+
+      this.addPokemonMarker(map, markerLat, markerLng);
+    });
+
+  }
+
+  addPokemonMarker(map, lat, lng) {
+    const randomPokemon = POKEMON_LIST[Math.floor(Math.random() * POKEMON_LIST.length)];
+    const icon = {
+      url: randomPokemon.marker_url,
+    };
+    const pokemonMarker = new google.maps.Marker({
+      position: map.getCenter(),
+      icon: icon,
       map: map
     });
 
-    marker.addListener('click', () => {
-      marker.setMap(null);
-      this.addMarker(map);
+    pokemonMarker.addListener('click', () => {
+      pokemonMarker.setMap(null);
+      this.openModal(randomPokemon);
 
-      if (this.props.onPokeballClick) {
-        this.props.onPokeballClick();
+      if (this.props.onPokemonClick) {
+        this.props.onPokemonClick(randomPokemon);
       }
     });
   }
 
+  openModal(pokemon) {
+    this.setState({
+      map: this.state.map,
+      pokemon: pokemon,
+      modalOpen: true
```

Comment:
You only need to update the state that's being updated in this case only this.setState ({ modalOpen: true }) is needed